ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1214,6 +1214,8 @@ func (r *RayClusterReconciler) calculateStatus(ctx context.Context, instance *ra
 	} else {
 		if utils.CheckAllPodsRunning(ctx, runtimePods) {
 			newInstance.Status.State = rayv1.Ready
+		} else {
+			newInstance.Status.State = rayv1.Unhealthy
```

Comment:
Would it be bad if cluster state went to unhealthy for larger ray cluster where only 1 node is down? I guess it depends on what is reading this state and if any action is taken.

Reply:
I wonder if it's safer to only check the head pod state if the state is already "Runnning"

Reply:
yea maybe it's safer for now to do that. Curious @kevin85421's thoughts

Reply:
I prefer to always check only the head pod even at the init stage. Because, for now, a serve health check will be added to all worker Pods' readiness probes if this RayCluster is created by RayService. However, not all worker Pods run a serve replica/proxy actor. Those idle worker Pods can have an `unready` status, causing the RayCluster status to be `unhealthy`.

Reply:
resolving since these changes are reverted in this PR